specification/ai/Azure.AI.Projects/datasets/routes.tsp

```diff
@@ -14,123 +15,46 @@ using Azure.Core.Foundations;
-    @path
-    version: string,
-  ): void;
-
-  #suppress "@azure-tools/typespec-azure-core/operation-missing-api-version"
-  #suppress "@azure-tools/typespec-azure-core/use-standard-operations"
-  @doc("Get dataset version.")
-  @route("/datasets/{name}/versions/{version}")
-  @get
-  getVersion(
-    @doc("Container name.")
-    @pattern("^[a-zA-Z0-9][a-zA-Z0-9\\-_]{0,254}$")
-    @path
-    name: string,
-
-    @doc("Version identifier.")
-    @path
-    version: string,
-  ): DatasetVersion;
-
-  #suppress "@azure-tools/typespec-azure-core/operation-missing-api-version"
-  #suppress "@azure-tools/typespec-azure-core/use-standard-operations"
-  @doc("Create or update version.")
-  @route("/datasets/{name}/versions/{version}")
-  @put
-  createOrUpdate(
-    @doc("Container name.")
-    @pattern("^[a-zA-Z0-9][a-zA-Z0-9\\-_]{0,254}$")
-    @path
-    name: string,
-
-    @doc("Version identifier.")
-    @path
-    version: string,
-
-    @doc("Version entity to create or update.")
-    @body
-    datasetVersion: DatasetVersion,
-  ): DatasetVersion;
-
-  #suppress "@azure-tools/typespec-azure-core/operation-missing-api-version"
-  #suppress "@azure-tools/typespec-azure-core/use-standard-operations"
-  @doc("Start pending upload.")
-  @post
-  @route("/datasets/{name}/versions/{version}/startPendingUpload")
-  createOrGetStartPendingUpload(
-    @doc("Container name.")
-    @pattern("^[a-zA-Z0-9][a-zA-Z0-9\\-_]{0,254}$")
-    @path
-    name: string,
-
-    @doc("Version identifier.")
-    @path
-    version: string,
-
-    @doc("Pensing upload request.")
-    @body
-    pendingUploadRequest: PendingUploadRequest,
-  ): PendingUploadResponse;
+alias ListDatasetVersionsParameters = {
+  @doc("Please choose OrderBy value from ['createdtime', 'modifiedtime']")
+  @query("$orderBy")
+  $orderBy?: string;
+
+  @doc("Top count of results, top count cannot be greater than the page size. If topCount > page size, results with be default page size count will be returned")
+  @query("$top")
+  $top?: int32;
+
+  @doc("Continuation token for pagination.")
+  @query("$skip")
+  $skip?: string;
+
+  @doc("Comma-separated list of tag names (and optionally values). Example: tag1,tag2=value2")
+  @query("$tags")
+  $tags?: string;
+
+  @doc("[ListViewType.ActiveOnly, ListViewType.ArchivedOnly, ListViewType.All] View type for including/excluding (for example) archived entities.")
+  @query("listViewType")
+  listViewType?: ListViewType;
+};
+
+alias ListLatestDataSetVersionParameters = {
+  @doc("View type for including/excluding (for example) archived entities.")
+  @query("listViewType")
+  listViewType?: ListViewType;
+};
+
+interface Datasets
+  extends ServicePatterns.VersionedOperations<
+      DatasetVersion,
+      ListLatestDataSetVersionParameters,
+      ListDatasetVersionsParameters
+    > {
+  /**
+   * Create or start a pending upload of a dataset
+   */
+  startPendingUpload is ServicePatterns.VersionedResourceAction<
```

Comment:
https://github.com/azure-sdk/azure-sdk-for-python/blob/49579936821bc59472c092ca426fc882afa410a6/sdk/ai/azure-ai-projects-dp1/azure/ai/projects/dp1/operations/_operations.py#L7742C1-L7742C107

Looks like the signature for this in autogenerated code is incorrect

Reply:
The `body` should be `PendingUploadRequest`, not `DatasetVersion`